pkg/gofr/gofr_test.go

```diff
@@ -561,3 +561,123 @@ func Test_AddCronJob_Success(t *testing.T) {
 }
+
+const indexHTML = "indexTest.html"
+
+func TestStaticHandler(t *testing.T) {
+	// Generating some files for testing
+	htmlContent := []byte("<html><head><title>Test Static File</title></head><body><p>Testing Static File</p></body></html>")
+
+	createPublicDirectory(t, htmlContent)
+
+	app := New()
+
+	app.httpRegistered = true
+	app.httpServer.port = 8022
+
+	go app.Run()
+	time.Sleep(1 * time.Second)
+
+	host := "http://localhost:8022"
+
+	tests := []struct {
+		desc                       string
+		method                     string
+		path                       string
+		statusCode                 int
+		expectedBody               string
+		expectedBodyLength         int
+		expectedResponseHeaderType string
+	}{
+		{
+			desc:                       "check file content index.html",
+			method:                     http.MethodGet,
+			path:                       "/" + defaultPublicStaticDir + "/" + indexHTML,
+			statusCode:                 http.StatusOK,
+			expectedBodyLength:         len(htmlContent),
+			expectedResponseHeaderType: "text/html; charset=utf-8",
+			expectedBody:               string(htmlContent),
+		},
+		{
+			desc:       "check public endpoint",
+			method:     http.MethodGet,
+			path:       "/" + defaultPublicStaticDir,
+			statusCode: http.StatusNotFound,
+		},
+	}
+
+	for i, tc := range tests {
+		request, err := http.NewRequestWithContext(context.Background(), tc.method, host+tc.path, http.NoBody)
+		if err != nil {
+			t.Fatalf("TEST[%d], Failed to create request, error: %s", i, err)
+		}
+
+		request.Header.Set("Content-Type", "application/json")
+
+		client := http.Client{}
+
+		resp, err := client.Do(request)
+		if err != nil {
+			t.Fatalf("TEST[%d], Request failed, error: %s", i, err)
+		}
+
+		bodyBytes, err := io.ReadAll(resp.Body)
+		if err != nil {
+			t.Fatalf("TEST[%d], Failed to read response body, error: %s", i, err)
+		}
+
+		body := string(bodyBytes)
+
+		assert.NoError(t, err, "TEST[%d], Failed.\n%s", i, tc.desc)
+		assert.Equal(t, tc.statusCode, resp.StatusCode, "TEST[%d], Failed with Status Body.\n%s", i, tc.desc)
+
+		if tc.expectedBody != "" {
+			assert.Contains(t, body, tc.expectedBody, "TEST[%d], Failed with Expected Body.\n%s", i, tc.desc)
+		}
+
+		if tc.expectedBodyLength != 0 {
+			contentLength := resp.Header.Get("Content-Length")
+			assert.Equal(t, strconv.Itoa(tc.expectedBodyLength), contentLength, "TEST[%d], Failed at Content-Length.\n%s", i, tc.desc)
+		}
+
+		if tc.expectedResponseHeaderType != "" {
+			assert.Equal(t,
+				tc.expectedResponseHeaderType,
+				resp.Header.Get("Content-Type"),
+				"TEST[%d], Failed at Expected Content-Type.\n%s", i, tc.desc)
+		}
+
+		resp.Body.Close()
+	}
+}
+
+func createPublicDirectory(t *testing.T, htmlContent []byte) {
+	t.Helper()
+
+	directory := "./" + defaultPublicStaticDir
+	if _, err := os.Stat(directory); err != nil {
+		if err = os.Mkdir("./"+defaultPublicStaticDir, 0755); err != nil {
```

Comment:
The preferred notation is `0o755`, but I'm fine.

Added since go1.13 https://go.dev/ref/spec#Integer_literals

Because of multiple misunderstanding

https://stackoverflow.com/questions/63877081/why-does-go-programming-treat-any-number-starting-with-zero-as-an-octal-and-how

https://github.com/golang/go/issues/33363

I'm just reporting this as I already wrote some comment.